Adding settings to plugin to provide username and email

Florian-Varrin opened the pull request:
Add gradle build system and add settings to the plugin.

Resolves #2 because if environment variable doesn't work, can be set via the settings.

Refactored the code to have separate `services`, `actions` and `settings`.

Removed `.jar` because can be build with gradle.

Florian-Varrin committed:
Add gradle build system

Florian-Varrin committed:
Add settings

Florian-Varrin committed:
Refactor code into services

Florian-Varrin committed:
Add input dialog if no username

Florian-Varrin committed:
Add clion version in build.gradle

Florian-Varrin committed:
Check for null header just in case

Eywek commented:
Thank you for the contribution!

But could we add back the `.jar` file? It could avoid every potential user to build it

Florian-Varrin committed:
Add .zip build file for installation.
Fix few issues and warnings
Add `automatically add header` option in config

Florian-Varrin commented:
@Eywek Yes I added the `distributions` directory to put the `.zip` build (it is now gradle does it).

I also added an option in the configuration to add automatically the header when creating files and a `README` to explain all that.
